src/Explain.h

```diff
@@ -32,199 +36,206 @@
 
 namespace souffle {
 
-class Explain {
+class ExplainConfig {
```

Comment:
ExplainConfig shouldn't be used outside of Explain, but the idea of ExplainConfig was so that we don't need to pass configuration around as arguments when constructing an Explain. Is there a better way to do this?

Reply:
Ah, that makes more sense, then.

Handling the output stream in ExplainConfig still seems problematic, though, and wasn't used in the constructor. If it is part of the configuration that will be passed around, then maybe use a getter/setter for it to handle deletion, or make it a unique_ptr. It already leaks memory when you change stream.